front/reservation.form.php

```diff
@@ -97,80 +97,16 @@
-    if (isset($_POST['resa']["end"])) {
-       // Compute dates to add.
-        $dates_to_add[$_POST['resa']["begin"]] = $_POST['resa']["end"];
-
-        if (
-            isset($_POST['periodicity']) && is_array($_POST['periodicity'])
-            && isset($_POST['periodicity']['type']) && !empty($_POST['periodicity']['type'])
-        ) {
-           // Compute others dates to add.
-            $dates_to_add += Reservation::computePeriodicities(
-                $_POST['resa']["begin"],
-                $_POST['resa']["end"],
-                $_POST['periodicity']
-            );
-        }
-    }
-   // Sort dates
-    ksort($dates_to_add);
-    if (
-        count($dates_to_add)
-        && isset($_POST['items'])
-        && count($_POST['items'])
-        && isset($_POST['users_id'])
-    ) {
-        foreach ($_POST['items'] as $reservationitems_id) {
-            $input                        = [];
-            $input['reservationitems_id'] = $reservationitems_id;
-            $input['comment']             = $_POST['comment'];
-
-            if (count($dates_to_add) > 1) {
-                $input['group'] = $rr->getUniqueGroupFor($reservationitems_id);
-            }
-            foreach ($dates_to_add as $begin => $end) {
-                $input['begin']    = $begin;
-                $input['end']      = $end;
-                $input['users_id'] = (int)$_POST['users_id'];
-
-                if (
-                    Session::haveRight("reservation", UPDATE)
-                    || (Session::getLoginUserID() === $input["users_id"])
-                ) {
-                    unset($rr->fields["id"]);
-                    if ($newID = $rr->add($input)) {
-                        Event::log(
-                            $newID,
-                            "reservation",
-                            4,
-                            "inventory",
-                            sprintf(
-                                __('%1$s adds the reservation %2$s for item %3$s'),
-                                $_SESSION["glpiname"],
-                                $newID,
-                                $reservationitems_id
-                            )
-                        );
-
-                        $rri = new ReservationItem();
-                        $rri->getFromDB($reservationitems_id);
-                        $item = new $rri->fields["itemtype"]();
-                        $item->getFromDB($rri->fields["items_id"]);
-
-                        Session::addMessageAfterRedirect(
-                            sprintf(
-                                __('Reservation added for item %s at %s'),
-                                $item->getLink(),
-                                Html::convDateTime($input['begin'])
-                            )
-                        );
-                    }
-                }
-            }
+    if (isset($_POST["items"])) {
+        if (isset($_POST['resa'])) {
```

Comment:
As we do not plan to add strict checks on all parameters yet, you should just call `Reservation::handleAddForm($_POST);` whatever the params are, and do nothing in this method if there is nothing to do.
Adding stricts checks on params would require many more work and we do not have yet an harmonized way to do it.

```suggestion
```